geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/membership/GMSJoinLeave.java

```diff
@@ -1456,12 +1456,18 @@ void processNetworkPartitionMessage(NetworkPartitionMessage<ID> msg) {
       return;
     }
     ID sender = msg.getSender();
-    if (getView().getMembers().contains(sender)) {
-      String str = "Membership coordinator " + msg.getSender()
-          + " has declared that a network partition has occurred";
-      forceDisconnect(str);
+
+    if (getView() != null && isJoined) {
+      if (getView().getMembers().contains(sender)) {
+        String str = "Membership coordinator " + msg.getSender()
+            + " has declared that a network partition has occurred";
+        forceDisconnect(str);
+      } else {
+        logger.warn("Ignoring the network partition message from a non-member: " + msg.getSender());
+      }
     } else {
-      logger.warn("Ignoring the network partition message from a non-member: " + msg.getSender());
+      logger.info(
+          "Ignoring, likely this message was intended for the previous Membership service... ");
```

Comment:
Not sure whether this should be logged at `info` level, even if users see it, they won't know what to do about it. If the `else` branch it's harmless in itself, I'd say we should just use `debug` level instead.

Reply:
wilco